yeat/workflows/snakefiles/Pacbio

```diff
@@ -10,73 +10,82 @@
 from shutil import copyfile
 
 
+def get_expected_files():
+    inputlist = []
+    for label in config["labels"]:
+        assembler = config["assemblers"][label]
+        for sample in config["label_to_samples"][label]:
+            inputlist.append(f"analysis/{sample}/{label}/{assembler}/quast/{sample}_report.html")
+    inputlist += expand("seq/fastqc/{sample}/{sample}_fastqc.html", sample=[*config["samples"]])
+    return inputlist
+
+
 rule all:
     input:
-        expand("analysis/quast/{assembler}/{sample}_report.html", assembler=config["assemblers"], sample=config["sample"]),
-        expand("seq/fastqc/{sample}_fastqc.html", sample=config["sample"])
+        get_expected_files()
 
 
 rule copyinput:
     output:
-        read="seq/input/{sample}.fq.gz"
+        read="seq/input/{sample}/{sample}.fq.gz"
     input:
-        read=config["fastq"]
+        lambda wildcards: config["samples"][wildcards.sample]
     run:
-        if input.read.endswith(".gz"):
-            copyfile(input.read, output.read)
+        if input[0].endswith(".gz"):
+            copyfile(input[0], output.read)
         else:
-            shell(f"gzip -c {input.read} > {output.read}")
+            shell(f"gzip -c {input[0]} > {output.read}")
```

Comment:
This code silently ignores any filenames except for the first. Is the expectation of one file per sample for Pacbio data enforced anywhere else in the code?